### PR TITLE
feat: new resource `aws_redshift_logging`

### PR DESCRIPTION
**Pull request by thulasirajkomminar**
`logging` block in `aws_redshift_cluster` is deprecated in favour of the new resource `aws_redshift_logging`

### CODE REVIEW

**Review by fatbasstard — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -85,21 +85,22 @@ variable "kms_key_arn" {
...
-  default     = true
-  description = "Enables logging information such as queries and connection attempts"
-}
+  type = object({
+    bucket_name           = optional(string, null)
+    bucket_lifecycle_rule = optional(any, [])
+    create_bucket         = optional(bool, true)
+    log_destination_type  = string
+    log_exports           = optional(list(string), ["connectionlog", "useractivitylog", "userlog"])
+    s3_key_prefix         = optional(string, "redshift-audit-logs/")
```
Comment (shoekstra) [inline variables.tf:95]: Can we keep this consistent with other bucket variables? Something like `bucket_key_prefix`?

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: main.tf [hunk 2]
```diff
@@ -1,6 +1,7 @@
 locals {
-  elastic_ip        = var.publicly_accessible ? aws_eip.default[0].public_ip : null
-  subnet_group_name = var.subnet_ids == null ? "default" : (var.redshift_subnet_group != null ? var.redshift_subnet_group : var.name)
+  create_logging_bucket = var.logging != null && try(var.logging.create_bucket, false) && try(var.logging.log_destination_type, "") == "s3" ? 1 : 0
```
Comment (shoekstra) [inline main.tf:2]: Do we need to compute a value here when we have `var.logging.create_bucket`?

Seems we should either use that or computer the value, not both.

Comment (thulasirajkomminar) [reply main.tf:2]: I wanted the flexibility to create the S3 bucket outside of this module(maybe a common log bucket where redshift will also use that).

File: variables.tf (hunk 1 above)
Comment (thulasirajkomminar) [reply variables.tf]: modified it

File: main.tf (hunk 2 above)
Comment (thulasirajkomminar) [reply main.tf]: changed the local variable logic

**Review by shoekstra — APPROVED** [review]

**Pull request merged**